src/generic_event_type.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+#ifndef CATA_SRC_GENERIC_EVENT_TYPE_H
+#define CATA_SRC_GENERIC_EVENT_TYPE_H
+
+#include <string>
+#include <climits>
+
+#include "bodypart.h"
+#include "field.h"
+#include "translations.h"
+#include "type_id.h"
+
+template<typename T>
+class generic_factory;
+
+const generic_event_type_id NEXT_WEATHER_GENERIC_EVENT( "next_weather" );
+
+struct generic_event_type_field {
+    field_type_str_id type;
+    int intensity;
+    time_duration age;
+    int radius;
+    bool outdoor_only;
+};
+
+struct spawn_type {
+    mtype_id target;
+    int target_range;
+    int hallucination_count;
+    int real_count;
+    int min_radius;
+    int max_radius;
+};
+
+struct effect_info {
+    time_duration length = 0_seconds;
+    int intensity = 0;
+    efftype_id id;
+    bodypart_str_id target_part;
+};
+
+struct generic_event_type {
```

Comment:
I agree with your naming dilemma.  Event already has two meanings in the codebase, it would be good not to add another.

Moreover, the term "event" is normally used for "a thing that happens which you might like to be informed of"; that's what the `event_bus` is all about.  That's not what you have here.  This is "a thing which we do to the game state, changing it in some way".  So event is definitely not the right name.

I was thinking "action", but then I realised that also has a meaning already.

Maybe "operation"?